### PR TITLE
evm: state transitions enabled params

### PR DESCRIPTION
**Pull request by fedekunze**

### CODE REVIEW

**Review by noot — APPROVED** [review]
looks good, but I'm wondering what the purpose of this is for?

**Comment by fedekunze** [comment]
> looks good, but I'm wondering what the purpose of this is for?

At main net launch, we will likely start the chain with the parameters disabled and then enable them when the chain is mature and has produced enough blocks. It's pretty standard in the community and when new chains are deployed (Cosmos Hub, Polkadot, etc)

File: x/evm/types/params.go [hunk 1]
```diff
@@ -66,8 +80,16 @@ func (p Params) Validate() error {
 func validateEVMDenom(i interface{}) error {
 	denom, ok := i.(string)
```
Comment (araskachoi) [inline x/evm/types/params.go:81]: this cheks and passes if the passed denom is a string but do need to check if it's the same as the one set in genesis?

**Review by araskachoi — APPROVED** [review]

File: x/evm/types/params.go (hunk 1 above)
Comment (fedekunze) [reply x/evm/types/params.go:81]: this is just local validation. We don't have access to the SDK context unfortunately from the parameters in order to test against other variables

**Pull request merged**